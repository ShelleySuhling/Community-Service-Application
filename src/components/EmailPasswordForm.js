import React, { Component } from 'react';
import { Button, Form, Segment } from 'semantic-ui-react'

class EmailPasswordForm extends Component {

    render() {
        let { confirmPassword, onSubmit, handleChange, error } = this.props
        return (
            <div className="login-container">
                {error ? <Segment inverted color='red' tertiary>{error}</Segment> : null}
                <Form>
                    <Form.Field>
                        <div className="login-form-label">Email</div>
                        <input placeholder='Email' type='email' onChange={handleChange} />
                    </Form.Field>
                    <Form.Field>
                        <div className="login-form-label">Password</div>
                        <input type='password' placeholder='Password' onChange={handleChange} />
                    </Form.Field>

                    {confirmPassword
                        ? <Form.Field>
                            <div className="login-form-label">Confirm Password</div>
                            <input type='password' placeholder='Password' onChange={handleChange} />
                        </Form.Field>
                        : null}

                    <Button type='submit' onClick={onSubmit}>Log In</Button>
                </Form>
            </div>
        )
    }
}

export default EmailPasswordForm