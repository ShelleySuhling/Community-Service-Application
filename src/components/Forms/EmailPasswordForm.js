import React, { Component } from 'react';
import { Button, Form, Segment } from 'semantic-ui-react'

class EmailPasswordForm extends Component {

    render() {
        let { confirmPassword, onSubmit, handleChange, error, isLoading, submitText } = this.props
        return (
            <div className="form-container">
                {error ? <Segment inverted color='red' tertiary>{error}</Segment> : null}
                <Form className={isLoading ? "loading" : null}>
                    <Form.Field>
                        <div className="login-form-label">Email</div>
                        <input placeholder='Email' type='email' onChange={handleChange} />
                    </Form.Field>
                    <Form.Field>
                        <div className="login-form-label">Password</div>
                        <input type='password' placeholder='Password' onChange={handleChange} />
                    </Form.Field>

                    {confirmPassword
                        ? <Form.Field>
                            <div className="login-form-label">Confirm Password</div>
                            <input type='password' placeholder='Password' onChange={handleChange} />
                        </Form.Field>
                        : null}

                    <div className="submit-item">
                        <Button type='submit' onClick={onSubmit}>{submitText}</Button>
                    </div>
                </Form>
            </div>
        )
    }
}

export default EmailPasswordForm