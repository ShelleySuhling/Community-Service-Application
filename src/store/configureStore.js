import { applyMiddleware, createStore } from 'redux'
import { persistStore, persistCombineReducers } from 'redux-persist';
import thunkMiddleware from 'redux-thunk'
import storage from 'redux-persist/lib/storage';

import rootReducer from '../reducers/rootReducer'

export default function configureStore(preloadedState) {

    const persistConfig = {
        key: 'root',
        storage: storage,
    };
    const pReducer = persistCombineReducers(persistConfig, rootReducer);

    const store = createStore(pReducer, applyMiddleware(thunkMiddleware));
    const persistor = persistStore(store);
    window.store = store;

    return { persistor, store }
}