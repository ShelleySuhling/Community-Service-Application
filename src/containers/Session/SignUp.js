import React, { Component } from 'react';
import { Redirect } from 'react-router-dom'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as sessionActions from '../../actions/sessionActions';
import PropTypes from 'prop-types';
import * as _ from 'lodash'

import EmailPasswordForm from '../../components/Forms/EmailPasswordForm'


class SignUp extends Component {

    componentWillUnmount() {
        this.props.sessionActions.clearAuthErrors()
    }

    handleChange = (event) => {
        this.setState({ [event.target.type]: event.target.value });
    }

    onSubmit = (event) => {
        this.props.sessionActions.requestSignUp(this.state.email, this.state.password)
    }


    render() {
        const { session } = this.props
        if (!_.isEmpty(session.user)) {
            return <Redirect to='/' />
        } else if (session.pending_signup) {
            return "loading"
        }
        else {
            return (
                <div className="content-container">
                    <EmailPasswordForm onSubmit={this.onSubmit} handleChange={this.handleChange} confirmPassword={true} error={session.error} />
                </div>
            )
        }
    }
}

SignUp.propTypes = {
    sessionActions: PropTypes.object,
    session: PropTypes.object
};

function mapStateToProps(state) {
    return {
        session: state.session,
    };
}
function mapDispatchToProps(dispatch) {
    return {
        sessionActions: bindActionCreators(sessionActions, dispatch)
    };
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SignUp);