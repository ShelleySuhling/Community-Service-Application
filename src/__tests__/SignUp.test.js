import * as React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate'

import EmailPasswordForm from '../components/Forms/EmailPasswordForm'
import SignUp from '../containers/Session/SignUp';

import configureStore from '../store/configureStore';
const { store } = configureStore()

describe('SignUp', () => {
    MockDate.set('1/1/2018');
    let submit = jest.fn();
    const signUpWrapper = shallow(<SignUp store={store} />).dive()
    const formWrapper = signUpWrapper.find(EmailPasswordForm).dive()
    it('matches snapshot', () => {
        expect(signUpWrapper).toMatchSnapshot();
    })
    it('renders EmailPasswordForm component', () => {
        expect(signUpWrapper.find(EmailPasswordForm).getElements()).toHaveLength(1)
    })
    it('correctly handles input change and changes state', () => {
        const eventObject = { target: { type: 'email', value: 'fakeemail@email.com' } }
        signUpWrapper.instance().handleChange(eventObject)
        expect(signUpWrapper.instance().state.email).toEqual('fakeemail@email.com')
    })
    it('correctly handles submitting form', () => {
        const request_sign_up_mock = jest.fn()
        signUpWrapper.instance().props.sessionActions.requestSignUp = request_sign_up_mock

        formWrapper.find({ type: 'email' }).simulate('change', { target: { value: "fakeemail@email.com", type: 'email' } })
        formWrapper.find({ type: 'password' }).simulate('change', { target: { value: "password", type: 'password' } })
        formWrapper.find({ type: 'password-confirm' }).simulate('change', { target: { value: "password", type: 'password-confirm' } })
        formWrapper.find({ type: 'submit' }).simulate('click', { preventDefault() { } })
        expect(request_sign_up_mock).toHaveBeenCalled()
    })
    it('unmounts and calls to clear auth errors', () => {
        const clear_auth_errors_mock = jest.fn()
        signUpWrapper.instance().props.sessionActions.clearAuthErrors = clear_auth_errors_mock

        signUpWrapper.instance().componentWillUnmount()
        expect(clear_auth_errors_mock).toHaveBeenCalled()
    })
    it('redirects to events page when user is logged in', () => {
        const redirect_to_events_mock = jest.fn()
        signUpWrapper.instance().redirectToEvents = redirect_to_events_mock
        signUpWrapper.instance().props.session.user = { id: 3 }

        signUpWrapper.instance().forceUpdate()
        expect(redirect_to_events_mock).toHaveBeenCalled()
    })
})