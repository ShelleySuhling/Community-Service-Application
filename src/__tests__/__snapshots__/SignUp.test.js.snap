// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    session={Object {}}
    sessionActions={
      Object {
        "clearAuthErrors": [Function],
        "requestSignIn": [Function],
        "requestSignOut": [Function],
        "requestSignUp": [Function],
        "updateSessionUser": [Function],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <EmailPasswordForm
        confirmPassword={true}
        handleChange={[Function]}
        onSubmit={[Function]}
        submitText="Sign Up"
      />,
      "className": "content-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "confirmPassword": true,
        "error": undefined,
        "handleChange": [Function],
        "isLoading": undefined,
        "onSubmit": [Function],
        "submitText": "Sign Up",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <EmailPasswordForm
          confirmPassword={true}
          handleChange={[Function]}
          onSubmit={[Function]}
          submitText="Sign Up"
        />,
        "className": "content-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "confirmPassword": true,
          "error": undefined,
          "handleChange": [Function],
          "isLoading": undefined,
          "onSubmit": [Function],
          "submitText": "Sign Up",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
